stripe.go

```diff
@@ -621,11 +631,37 @@ func (s *BackendImplementation) shouldRetry(err error, req *http.Request, resp *
 
 	stripeErr, _ := err.(*Error)
 
-	// TODO: This retries any non-Stripe errors produced as part of the
-	// communication process. It generally works, but there are many errors
-	// that should *not* be retried. Try to make this more granular by
-	// including only connection errors, timeout errors, etc.
+	// Don't retry if the context was was cancelled or its deadline was
+	// exceeded.
+	if req.Context() != nil && req.Context().Err() != nil {
+		return false
```

Comment:
Should we log more clearly here?

Reply:
Hmm, good point, but can we do this in a follow up PR? I think it might be a good idea to produce a good logging message for all nixed retries.